### PR TITLE
MS16206 - Enable flight by default in desktop, controls menu changes re: flying

### PR DESCRIPTION
**Pull request by r3tk0n**
Addresses Manuscript #16206. Flying should always be enabled in desktop mode (unless overridden by a script or something).

Flying should be off for users who launch in HMD on their first run.

The "Flying & jumping" menu item has been removed from the "Controls..." menu in desktop mode, but remains if accessed via the tablet in VR mode.

### CODE REVIEW

File: interface/src/ui/PreferencesDialog.cpp [hunk 1]
```diff
@@ -266,25 +266,61 @@ void setupPreferences() {
...
         auto getter = [=]()->bool { return myAvatar->useAdvancedMovementControls(); };
         auto setter = [=](bool value) { myAvatar->setUseAdvancedMovementControls(value); };
         preferences->addPreference(new CheckPreference(MOVEMENT,
+            QStringLiteral("Advanced movement for hand controllers"),
+            getter, setter));
+    }
+    {
+        auto getter = [=]()->int { return myAvatar->getSnapTurn() ? 0 : 1; };
+        auto setter = [=](int value) { myAvatar->setSnapTurn(value == 0); };
+        auto preference = new RadioButtonsPreference(MOVEMENT, "Snap turn / Smooth turn", getter, setter);
```
Comment (wayne-chen) [inline interface/src/ui/PreferencesDialog.cpp]: I don't think this is needed.  This is out of the scope of the MS ticket.

File: interface/src/ui/PreferencesDialog.cpp [hunk 2]
```diff
@@ -266,25 +266,61 @@ void setupPreferences() {
...
+        QStringList items;
+        items << "Snap turn" << "Smooth turn";
+        preference->setItems(items);
+        preferences->addPreference(preference);
+    }
+    {
+        auto getter = [=]()->float { return myAvatar->getUserHeight(); };
+        auto setter = [=](float value) { myAvatar->setUserHeight(value); };
+        auto preference = new SpinnerPreference(MOVEMENT, "User real-world height (meters)", getter, setter);
```
Comment (wayne-chen) [inline interface/src/ui/PreferencesDialog.cpp]: This too.

File: interface/src/ui/PreferencesDialog.cpp [hunk 3]
```diff
@@ -266,25 +266,61 @@ void setupPreferences() {
...
+        preference->setMin(1.0f);
+        preference->setMax(2.2f);
+        preference->setDecimals(3);
+        preference->setStep(0.001f);
+        preferences->addPreference(preference);
+    }
+    {
+        auto preference = new ButtonPreference(MOVEMENT, "RESET SENSORS", [] {
```
Comment (wayne-chen) [inline interface/src/ui/PreferencesDialog.cpp]: I don't know if this is needed either.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11432/HighFidelity-Beta-PR13652-dfcf47e.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11432/HighFidelity-Beta-PR13652-dfcf47e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11432/HighFidelity-Beta-Interface-PR13652-dfcf47e.exe

**Comment by r3tk0n** [comment]
**Test Plan:**

_Goal:_ Verify the following:
1.)  Flying is now enabled by default for desktop users.
2.) "Flying & jumping" menu item has been renamed to "Flying"
3.) "Flying" menu item has been removed from "Settings > Controls..." menu in desktop mode.
4.) "Flying" menu item remains in "Tablet > Menu > Settings > Controls..." menu in HMD mode.

_Requirements:_ HMD, fresh install with no settings stored (purge %appdata% HiFi directories).

_Steps:_
- Start Interface.exe in desktop mode.
- Attempt to fly. If successful, Goal 1 is good.
- Navigate to "Settings > Controls..." in desktop mode
- If "Flying" checkbox is not listed, Goal 3 is good.
- Close Interface.exe and purge settings.
- Start Interface.exe in HMD mode.
- Navigate to "Tablet > Menu > Settings > Controls..." menu via tablet in HMD mode.
- If "Flying" checkbox is present (not "Flying & jumping") then goals 2 and 4 are good.

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Fail @qliacoombs @Qliemillar 
Unable to fly in desktop mode by default
In HMD the "Flying" option still affects jumping. Unable to jump be default.

[hifi-log.txt](https://github.com/highfidelity/hifi/files/2214310/hifi-log.txt)
[hifi-logdesk.txt](https://github.com/highfidelity/hifi/files/2214321/hifi-logdesk.txt)

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/11454/HighFidelity-Beta-PR13652-3108d65.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11454/HighFidelity-Beta-PR13652-3108d65.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11454/HighFidelity-Beta-Interface-PR13652-3108d65.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11454/HighFidelity-Beta-PR13652-3108d65.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11454/HighFidelity-Beta-Interface-PR13652-3108d65.exe

**Comment by r3tk0n** [comment]
**Test Plan:**

_Goal:_ Verify the following:
1.) Flying is now enabled by default for desktop users.
2.) "Flying & jumping" menu item has been removed from "Settings > Controls..." menu in desktop mode.
3.) "Flying & jumping" menu item remains in "Tablet > Menu > Settings > Controls..." menu in HMD mode.

_Requirements:_ HMD, fresh install with no settings stored (purge %appdata% HiFi directories).

_Steps:_

- Start Interface.exe in desktop mode.
- Attempt to fly. If successful, Goal 1 is good.
- Navigate to "Settings > Controls..." in desktop mode
- If "Flying & jumping" checkbox is not listed, Goal 2 is good.
- Close Interface.exe and purge settings.
- Start Interface.exe in HMD mode.
- Navigate to "Tablet > Menu > Settings > Controls..." menu via tablet in HMD mode.
- If "Flying & jumping" checkbox is present then goal 3 is good.

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Fail @Qliemillar @qliacoombs 
Goal 1 fails. Unable to fly by default in desktop mode. Still able to jump as expected, flying however is disabled.

**Comment by r3tk0n** [comment]
@qlimnavarro That was with a clean install, all settings purged from %appdata%?

**Comment by qlimnavarro** [comment]
@r3tk0n Yes, I uninstalled and reinstalled purging all data within local and roaming in appdata.

**Comment by wayne-chen** [comment]
@qlimnavarro can you please re-test this on your system with a fresh download of the latest?  Does this fail on all operating systems?  It seems to have passed on my end - I am able to fly in desktop mode upon first opening the application in desktop mode.

**Comment by qlimnavarro** [comment]
@wayne-chen @r3tk0n Fresh install on Mac, Windows, and Android. All opened in desktop mode and all 3 platforms can jump but not fly. Tested using PR13652-3108d65. 

Log and video attached: 
[Fly.zip](https://github.com/highfidelity/hifi/files/2215457/Fly.zip)

**Comment by r3tk0n** [comment]
@qlimnavarro After re-testing on this end saw the behavior. Wayne and I are testing a fix now, more to come.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11471/HighFidelity-Beta-PR13652-bc91aff.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/11471/HighFidelity-Beta-Interface-PR13652-bc91aff.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11471/HighFidelity-Beta-PR13652-bc91aff.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/11471/HighFidelity-Beta-Interface-PR13652-bc91aff.exe

**Comment by r3tk0n** [comment]
@qlimnavarro This one's good to go, same test plan as before.

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Test plan passes and default flying and jumping settings are correct on all platforms.

**Pull request merged**